slick-hikaricp/src/main/scala/slick/jdbc/hikaricp/HikariCPJdbcDataSource.scala

```diff
@@ -24,36 +24,62 @@ object HikariCPJdbcDataSource extends JdbcDataSourceFactory {
   def forConfig(c: Config, driver: Driver, name: String, classLoader: ClassLoader): HikariCPJdbcDataSource = {
     val hconf = new HikariConfig()
 
-    // Connection settings
-    if (c.hasPath("dataSourceClass")) {
-      hconf.setDataSourceClassName(c.getString("dataSourceClass"))
-    } else {
-      Option(c.getStringOr("driverClassName", c.getStringOr("driver"))).map(hconf.setDriverClassName _)
+    // Essential settings
+
+    // Use HikariCP `dataSourceClassName` as the main configuration and fallback to
+    // `dataSourceClass`, `driverClassName` and finally `drive`.
+    c.getStringOpt("dataSourceClassName").orElse(c.getStringOpt("dataSourceClass")) match {
+      case Some(className) => hconf.setDataSourceClassName(className)
+      case None => c.getStringOpt("driverClassName").orElse(c.getStringOpt("driver")).foreach(hconf.setDriverClassName)
     }
-    hconf.setJdbcUrl(c.getStringOr("url", null))
-    c.getStringOpt("user").foreach(hconf.setUsername)
+
+    // Use `jdbcUrl` an then `url` to configure the pool. According to HikariCP docs, when
+    // using this property with "old" drivers, you may also need to set the driverClassName
+    // property.
+    c.getStringOpt("jdbcUrl").orElse(c.getStringOpt("url")).foreach(hconf.setJdbcUrl)
+    c.getStringOpt("username").orElse(c.getStringOpt("user")).foreach(hconf.setUsername)
     c.getStringOpt("password").foreach(hconf.setPassword)
+
     c.getPropertiesOpt("properties").foreach(hconf.setDataSourceProperties)
 
-    // Pool configuration
+    // Frequently used pool configuration
+    c.getBooleanOpt("autoCommit").foreach(hconf.setAutoCommit)
+
+    val numThreads = c.getIntOr("numThreads", 20)
+
     hconf.setConnectionTimeout(c.getMillisecondsOr("connectionTimeout", 1000))
```

Comment:
This is one of the places where Slick is not using HikariCP defaults. While Slick uses 1 second, Hikari defaults to 30 seconds.

Should Slick respect HikariCP default?